Sources/SKSupport/Platform.swift

```diff
@@ -10,10 +10,74 @@
 //
 //===----------------------------------------------------------------------===//
 
-import enum TSCUtility.Platform
+import Foundation
+
+import class TSCBasic.Process
+import struct TSCBasic.AbsolutePath
+import var TSCBasic.localFileSystem
+
+private func isAndroid() -> Bool {
+  return (try? localFileSystem.isFile(AbsolutePath(validating: "/system/bin/toolchain"))) ?? false ||
+      (try? localFileSystem.isFile(AbsolutePath(validating: "/system/bin/toybox"))) ?? false
+}
+
+private func isDebian() -> Bool {
+  return (try? localFileSystem.isFile(AbsolutePath(validating: "/etc/debian_version"))) ?? false
+}
+
+private func isFedora() -> Bool {
+  if (try? localFileSystem.isFile(AbsolutePath(validating: "/etc/redhat-release"))) ?? false {
+    return true
+  }
+
+  if (try? localFileSystem.isFile(AbsolutePath(validating: "/etc/centos-release"))) ?? false {
+    return true
+  }
+
+  if (try? localFileSystem.isFile(AbsolutePath(validating: "/etc/fedora-release"))) ?? false {
+    return true
+  }
+
+  return (try? localFileSystem.readFileContents(AbsolutePath(validating: "/etc/system-release")))?.cString.hasPrefix("Amazon Linux release 2") ?? false
+}
+
+public enum Platform: Equatable {
+  public enum LinuxDistribution: Equatable {
+    case debian
+    case fedora
+  }
+
+  case android
+  case darwin
+  case linux(LinuxDistribution)
+  case windows
+}
 
 extension Platform {
+  // This is not just a computed property because the ToolchainRegistryTests
+  // change the value.
+  public static var current: Platform? = {
+    #if os(Windows)
+    return .windows
+    #else
+    switch try? Process.checkNonZeroExit(args: "uname")
+        .trimmingCharacters(in: .whitespacesAndNewlines)
+        .lowercased() {
+    case .some("darwin"):
+      return .darwin
+    case .some("linux"):
```

Comment:
Nit: I think this reads better as

```suggestion
    case "darwin"?:
      return .darwin
    case "linux"?:
```

Reply:
I have to agree, thanks!